lib/qr_code/format_version.ex

```diff
@@ -96,82 +96,85 @@ defmodule QRCode.FormatVersion do
 
   @spec put_information(QR.t()) :: Result.t(String.t(), QR.t())
   def put_information(
-        %QR{matrix: matrix, version: version, ecc_level: :low, mask_num: mask_num} = qr
+        %QR{matrix: matrix, version: version, ecc_level: ecc_level, mask_num: mask_num} = qr
       )
-      when masking(mask_num) and version(version) and version < 7 do
+      when masking(mask_num) and version(version) do
     matrix
-    |> Result.map(&set_format_info(&1, @low, mask_num, version))
-    |> Result.map(fn matrix -> %{qr | matrix: matrix} end)
+    |> set_format_info(ecc_level, mask_num, version)
+    |> Result.map(&set_version_info(&1, version))
+    |> Result.and_then(fn matrix -> %{qr | matrix: matrix} end)
```

Comment:
Why is here `and_then`? Supplied lambda function does not return `{:ok, ...}` or `{:error, ...}` value.